uvicorn/supervisors/statreload.py

```diff
@@ -8,10 +9,15 @@
 
 
 class StatReload(BaseReload):
+    ignored_file_regexes = r"\.py[cod]$", r"\.___jb_...___$", r"\.sw.$", "~$"
```

Comment:
It’s not clear to what all of these are?

It might be worth having a look into what gunicorn or Django devserver do for reload behavious, since this seems a bit ad hoc at the moment?

Reply:
from a quick look at it it seems gunicorn's reloader will only react to `.py` files:

```
    def get_files(self):
        fnames = [
            COMPILED_EXT_RE.sub('py', module.__file__)
            for module in tuple(sys.modules.values())
            if getattr(module, '__file__', None)
        ]


        fnames.extend(self._extra_files)


        return fnames
```

it get all files to watch by looking at `sys.modules.values()` and uses the regex
[`r'py[co]$'` ](https://github.com/benoitc/gunicorn/blob/5425af8941e1d43c57b5b2fb3fbd7f728688ce01/gunicorn/reloader.py#L14) to transform all the matched extensions (pyc, py and pyo, which is almost the 1st ignored regex above by the way) to `.py`

the above regexes are taken from what watchgod does in its DefaultWatcher:

1. `r"\.py[cod]$"` wil ignore all `.pyc` and `.pyo` and `.pyd`

1. `r"\.___jb_...___$"` honestly I dont know

1. `r"\.sw.$"` this should be swap files, handy for vim users

1. `"~$"` same category as above I suppose, vim backup files

Reply:
Drive-by comment: `"~$"` represents backup files produced by emacs.